### PR TITLE
url audio

### PR DESCRIPTION
**Pull request by liyiheng**
mediaPlayer.setDataSource("url");

### CODE REVIEW

File: rxandroidaudio/src/main/java/com/github/piasy/rxandroidaudio/RxAudioPlayer.java [hunk 1]
```diff
@@ -131,6 +133,30 @@ public void call(final SingleSubscriber<? super Boolean> singleSubscriber) {
                     }
                 }
             });
+        } else if (config.mType == PlayConfig.TYPE_URL && config.url != null &&
```
Comment (Piasy) [inline rxandroidaudio/src/main/java/com/github/piasy/rxandroidaudio/RxAudioPlayer.java]: Thanks for your PR!

Maybe we can turn this if-else-if into switch, and do some common check before it.

And you can check url with `TextUtil.isEmpty`.

File: rxandroidaudio/src/main/java/com/github/piasy/rxandroidaudio/PlayConfig.java [hunk 2]
```diff
@@ -28,6 +30,8 @@
 
     final File mAudioFile;
 
+    final String url;
```
Comment (Piasy) [inline rxandroidaudio/src/main/java/com/github/piasy/rxandroidaudio/PlayConfig.java]: And could you please rename this field into `mUrl`?

**Review by Piasy — CHANGES_REQUESTED** [review]

File: rxandroidaudio/src/main/java/com/github/piasy/rxandroidaudio/PlayConfig.java [hunk 3]
```diff
@@ -76,6 +88,8 @@ public static Builder res(Context context, @RawRes int audioResource) {
 
         File mAudioFile;
 
+        String url;
```
Comment (Piasy) [inline rxandroidaudio/src/main/java/com/github/piasy/rxandroidaudio/PlayConfig.java]: This field please.

File: rxandroidaudio/src/main/java/com/github/piasy/rxandroidaudio/RxAudioPlayer.java [hunk 4]
```diff
@@ -86,54 +89,86 @@ public int progress() {
      * play audio from local file. should be scheduled in IO thread.
      */
     public Single<Boolean> play(@NonNull final PlayConfig config) {
-        if (config.mType == PlayConfig.TYPE_FILE && config.mAudioFile != null &&
-                config.mAudioFile.exists()) {
-            return Single.create(new Single.OnSubscribe<Boolean>() {
-                @Override
-                public void call(final SingleSubscriber<? super Boolean> singleSubscriber) {
-                    stopPlay();
+        switch (config.mType) {
```
Comment (Piasy) [inline rxandroidaudio/src/main/java/com/github/piasy/rxandroidaudio/RxAudioPlayer.java:92]: I'm just wondering whether we could encapsulate the parameter check into `PlayConfig` it self, e.g. `isArgumentIllegal()`, so we don't need check it inside the switch, and no more fall through in each case.

If you don't want to do this update, it's totally ok, I'll clean it up and cut a new release :)

**Review by Piasy — CHANGES_REQUESTED** [review]

File: rxandroidaudio/src/main/java/com/github/piasy/rxandroidaudio/PlayConfig.java (hunk 3 above)
Comment (liyiheng) [reply rxandroidaudio/src/main/java/com/github/piasy/rxandroidaudio/PlayConfig.java]: okay

File: rxandroidaudio/src/main/java/com/github/piasy/rxandroidaudio/RxAudioPlayer.java (hunk 4 above)
Comment (liyiheng) [reply rxandroidaudio/src/main/java/com/github/piasy/rxandroidaudio/RxAudioPlayer.java:92]: I didn't clone the  repository, just downloaded the zip file.
So I have to commit on the web file by file. It's great if you can do this update.

In my opinion, the parameter check in  `PlayConfig` will make the code clean, but there wolud be two `switch(mType)` , one in `PlayConfig` , and one in `play()`.
Maybe `isArgumentIllegal()` can be implemented as 3 methods.

Comment (Piasy) [reply rxandroidaudio/src/main/java/com/github/piasy/rxandroidaudio/RxAudioPlayer.java:92]: No problem, I think two switch is ok.

**Comment by Piasy** [comment]
@XanthusL I've just update two commits, and cut `v1.4.0`.

**Pull request merged**